src/games/strategy/sound/ClipPlayer.java

```diff
@@ -442,239 +419,67 @@ private void parseClipPaths(final String pathName, final boolean subFolder) {
     } catch (final Exception e2) {
       thisSoundFile = null;
     }
-    if (thisSoundFile == null || !thisSoundFile.exists()) {
-      // final long startTime = System.currentTimeMillis();
-      // we are probably using zipped sounds. there might be a better way to do this...
-      final String soundFilePath = thisSoundURL.getPath();
-      if (soundFilePath != null && soundFilePath.length() > 5 && soundFilePath.indexOf(".zip!") != -1) {
-        // so the URL with a zip or jar in it, will start with "file:", and unfortunately when you make a file and test
-        // if it exists, if it
-        // starts with that it doesn't exist
-        final int index1 = Math.max(0, Math.min(soundFilePath.length(),
```

Comment:
All of this code is never executed. Debugging the what remains, it looks like the only active code in this whole method is this:
```
      if (testClipSuccessful(thisSoundURL)) {
        availableSounds.add(thisSoundURL);
      }
```
For the wav files in a map zip case, the parameter value will look something like this:
```
jar:file:/home/dan/triplea/maps/star_wars_galactic_war.zip!/sounds/click2.wav
```
The code in CiipPlayer then has no problem creating a Clip object with that.

I would surmise then that it was never realized that this code was all dead code effectively, that the fall back logic executed later was the only logic that was really doing anything.

For reference, debugging this, a stack trace to get to this point looks like:
```
ClipPlayer.createAndAddClips(String) line: 403	
ClipPlayer.parseClipPaths(String, boolean) line: 378	
ClipPlayer.loadClipPath(String, boolean, boolean) line: 333	
ClipPlayer.loadClip(String, String, boolean) line: 322	
ClipPlayer.preLoadClip(String) line: 303	
SoundPath.preLoadSounds(SoundPath$SoundType) line: 144	
ClipPlayer.getInstance(ResourceLoader, GameData) line: 140	
UIContext$1.run() line: 98	
Thread.run() line: 745
```

Stack trace with method signatures and a bit more context:
```
      SoundPath.preLoadSounds(SoundPath.SoundType.GENERAL);
        clipPlayer.preLoadClip(CLIP_CLICK_PLOT);

  protected void preLoadClip(final String clipName) {
    loadClip(clipName, null, true);

  private synchronized Clip loadClip(final String clipName, final String subFolder, final boolean parseThenTestOnly) {
        return loadClipPath(clipName, false, parseThenTestOnly);

  private Clip loadClipPath(final String pathName, final boolean subFolder, final boolean parseThenTestOnly) {
    if (!sounds.containsKey(pathName)) {
      // parse sounds for the first time
      parseClipPaths(pathName, subFolder);

  private void parseClipPaths(final String pathName, final boolean subFolder) {
      :
      availableSounds.addAll(createAndAddClips(ASSETS_SOUNDS_FOLDER + "/" + path));


  protected List<URL> createAndAddClips(final String resourceAndPathURL) {
   
```

And then eventually it looks like most of the time the code short circuits searching for possible paths, and those that do exist, then eventually hit the "testClipSuccessful" logic which is able to load the sound file, even when inside of a compressed file.